### PR TITLE
Add caching to clickhouse endpoints

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- previously caching only applied to postgres queries
- now the clickhouse endpoints use the cache decorator too
- TODO: follow up ticket to consolidate how funnel caching is handled

*If this affects the frontend, include screenshots.*  

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: posthog/tasks/update_cache.py
Comment (timgl) [inline posthog/tasks/update_cache.py:103]: Couple things about this file
1. It doesn't get called on ee yet in posthog/celery.py, so we'll need to change the conditional there
2. I'm worried this will again completely clog clickhouse queries. I had a slightly better solution for cohorts (once every minute, only do a handful at a time) but it's still not perfect.
3. Be good to have tests

Comment (EDsCODE) [reply posthog/tasks/update_cache.py:103]: Got it! Thanks for the heads up. I have a weird habit of marking the PR as ready for review when an initial implementation is in but didn't finish tests yet sorry for premature request if that's what it seemed like

File: posthog/models/filters/retention_filter.py
Comment (EDsCODE) [inline posthog/models/filters/retention_filter.py:65]: Had to fix date handling in retention filter so that when the celery tasks create the filter, the dates were the same as the initial filter

File: posthog/tasks/test/test_update_cache.py
Comment (EDsCODE) [inline posthog/tasks/test/test_update_cache.py:71]: ensures that the retention/trend/stickiness filters trigger a proper dashboarditem refresh

File: posthog/models/filters/filter.py
Comment (EDsCODE) [inline posthog/models/filters/filter.py:134]: not DRY. Going to change how this is done in follow-up PR that focuses on the filters

File: posthog/tasks/update_cache.py
Comment (timgl) [inline posthog/tasks/update_cache.py:75]: This needs to be filtered by `refreshing=False` and maybe also `last_refresh__lte=timezone.now() - relativedelta(minutes=MAX_AGE_MINUTES)`

**Review by timgl — APPROVED** [review]
Looks good apart from the comment on the task. I also think we should closely monitor database load and be ready to revert if we see load go up

**Comment by EDsCODE** [comment]
I'll wait until some of the infra issues are stablized!

**Review by timgl — COMMENTED** [review]
Let's hold off merging this, I think I've got a slightly better solution

**Comment by EDsCODE** [comment]
Going to handle broken tests after #2736. The issue is with the changes needed for the retention filter. It'll be easier to fix with the mixins rather than working through it now just to have to remove those changes once the filter factor is merged

**Comment by EDsCODE** [comment]
Closing this as it's partially been handled in #2745 . Will recheck this after #2736

**Pull request closed without merging**
